<launch>
    <arg name="neural_network_inference_type"/> <!-- Options: cpu, torch_gpu or trt_gpu -->
    <arg name="audio_analyzer_voice_probability_threshold"/>
    <arg name="audio_analyzer_class_probability_threshold"/>
    <arg name="audio_analyzer_perception_delay"/>

    <node pkg="audio_analyzer" type="audio_analyzer_node.py" name="audio_analyzer_node" launch-prefix="bash -c 'sleep $(arg audio_analyzer_perception_delay); $0 $@' ">
        <param name="inference_type" value="$(arg neural_network_inference_type)"/>
        <param name="interval" value="8000"/>
        <param name="voice_probability_threshold" value="$(arg audio_analyzer_voice_probability_threshold)"/>
        <param name="class_probability_threshold" value="$(arg audio_analyzer_class_probability_threshold)"/>

        <remap from="audio_in" to="audio_input_signed_16_16000"/>
        <remap from="audio_in/filter_state" to="audio_analyzer/filter_state"/>
        <remap from="sst" to="audio_sst"/>
    </node>

</launch>
